test/dlc_tests/sanity/test_pre_release.py

```diff
@@ -184,6 +184,53 @@ def test_tf_serving_version_cpu(tensorflow_inference):
 
     stop_and_remove_container(container_name, ctx)
 
+# Added by hballuru to confirm tf-serving-api version is matching tf serving version
+@pytest.mark.usefixtures("sagemaker")
+@pytest.mark.model("N/A")
+@pytest.mark.canary("Run non-gpu tf serving version test regularly on production images")
+def test_tf_serving_api_version_cpu(tensorflow_inference):
+    """
+    For non-huggingface non-GPU TF inference images, check that the tag version matches the version of TF serving api
+    in the container.
+
+    Huggingface includes MMS and core TF, hence the versioning scheme is based off of the underlying tensorflow
+    framework version, rather than the TF serving version.
+
+    GPU inference images will be tested along side `test_framework_and_cuda_version_gpu` in order to be judicious
+    about GPU resources. This test can run directly on the host, and thus does not require additional resources
+    to be spun up.
+
+    @param tensorflow_inference: ECR image URI
+    """
+    # Set local variable to clarify contents of fixture
+    image = tensorflow_inference
+
+    if "gpu" in image:
+        pytest.skip(
+            "GPU images will have their framework version tested in test_framework_and_cuda_version_gpu")
```

Comment:
I read the comment above saying `This test can run directly on the host, and thus does not require additional resources to be spun up.` . If that is the case, why do we want to tackle GPU test separately. Why can't we allow this test to run for GPU images as well. I think that this test does not spawn any ec2 instance and it would be judicious if we use the same test for GPU images.

Reply:
the code has been moved away from the method